### PR TITLE
(BKR-321) Added a LXC driver for beaker

### PR DESCRIPTION
**Pull request by ashish1099**
Added a lxc drive for beaker. working properly as expected.
Need to add ruby-lxc gem in gemfile

### CODE REVIEW

**Comment by anodelman** [comment]
Would you be willing to become a maintained for this code - we are unable to test it internally.

**Comment by ashish1099** [comment]
@anodelman yep sure. We are using it internally. we were facing lot of problem with docker and systemd inside container. lot of hack we had to get the thing working, and lxc we have been using so wrote one for it.

File: lib/beaker/hypervisor/lxc.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      container.save_config
+
+      @logger.notify "Starting Lxc #{host}"
+      container.start
+
+      @logger.notify "Attaching Lxc to #{host}"
+        container.attach(:wait => true) do
+          case host['platform']
+          when /ubuntu/, /debian/
+            puts `apt-get update`
```
Comment (anodelman) [inline lib/beaker/hypervisor/lxc.rb]: I'm a little confused by the use of 'puts' here - is this just to print the result of the command?  Is this command to be run on node executing beaker, or on a lxc container?

**Comment by anodelman** [comment]
Please add the new dependency 'ruby-lxc' to the beaker.gemspec.

**Comment by anodelman** [comment]
Also, to meet our contributor requirements each commit needs to be associated with a beaker ticket number like (BKR-xxx) - beaker tickets can be filed here: https://tickets.puppetlabs.com/browse/BKR/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel

**Comment by ashish1099** [comment]
Created a ticket https://tickets.puppetlabs.com/browse/BKR-321

File: lib/beaker/hypervisor/lxc.rb (hunk 1 above)
Comment (ashish1099) [inline lib/beaker/hypervisor/lxc.rb]: @anodelman nope, this will run inside container, #41 line show that its getting attach to the container.

**Comment by anodelman** [comment]
You need to update your commit message with (BKR-321).

**Comment by anodelman** [comment]
Last comment still current.

**Comment by ashish1099** [comment]
@anodelman will update today :)

**Comment by anodelman** [comment]
Still waiting on commit message update.

**Comment by ashish1099** [comment]
I will raise a new. i somehow screwed the whole thing..

**Pull request closed without merging**